### PR TITLE
Issue/4056 eol shard

### PR DESCRIPTION
**Pull request by fulmicoton**

### CODE REVIEW

File: quickwit/quickwit-indexing/src/source/ingest/mod.rs [hunk 1]
```diff
@@ -135,6 +137,7 @@ pub struct IngestSource {
     fetch_stream: MultiFetchStream,
     publish_lock: PublishLock,
     publish_token: PublishToken,
+    shard_positions: FnvHashMap<ShardId, Position>,
```
Comment (guilload) [inline quickwit/quickwit-indexing/src/source/ingest/mod.rs:140]: Why not reuse the `assigned_shards: HashMap<ShardId, AssignedShard>` to track the publish position?

File: quickwit/quickwit-indexing/src/source/ingest/mod.rs [hunk 2]
```diff
@@ -466,14 +494,33 @@ impl Source for IngestSource {
...
             "assigned_shards": self.assigned_shards.keys().copied().collect::<Vec<ShardId>>(),
             "publish_token": self.publish_token,
+            "shard_states": shard_states,
         })
     }
 }
 
+#[derive(Serialize, Deserialize, Copy, Clone, Debug)]
+#[serde(rename_all = "snake_case")]
+enum ShardIngestionState {
```
Comment (guilload) [inline quickwit/quickwit-indexing/src/source/ingest/mod.rs:519]: There's already an `enum IndexingStatus` per shard in `assigned_shards`.

File: quickwit/quickwit-indexing/src/source/ingest/mod.rs (hunk 1 above)
Comment (fulmicoton) [reply quickwit/quickwit-indexing/src/source/ingest/mod.rs:140]: ah yes that's perfect! I am glad I had forgotten to put that PR in draft  :)

**Comment by fulmicoton** [comment]
:-/ this PR is broken because it publish completed as soon as all of the source has been consumed.
Not as soon as the docs have been published.

**Pull request closed without merging**